cmd/virt-launcher-monitor/virt-launcher-monitor.go

```diff
@@ -0,0 +1,306 @@
+
+	for _, file := range files {
+		err = os.RemoveAll(file)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func terminateIstioProxy() {
+	httpClient := &http.Client{Timeout: httpRequestTimeout}
+	if istioProxyPresent(httpClient) {
+		isRetriable := func(err error) bool {
+			if net.IsConnectionReset(err) || net.IsConnectionRefused(err) || k8serrors.IsServiceUnavailable(err) {
+				return true
+			}
+			return false
+		}
+		err := retry.OnError(retry.DefaultBackoff, isRetriable, func() error {
+			resp, err := httpClient.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", envoyMergedPrometheusTelemetryPort), "", nil)
+			if err != nil {
+				log.Log.Reason(err).Error("failed to request istio-proxy termination, retrying...")
+				return err
+			}
+
+			defer resp.Body.Close()
+			if resp.StatusCode != http.StatusOK {
+				log.Log.Errorf("status code received: %d", resp.StatusCode)
+				return err
+			}
+
+			return nil
+		})
+		if err != nil {
+			log.Log.Reason(err).Error("all attempts to terminate istio-proxy failed")
+		}
+	}
+}
+
+func istioProxyPresent(httpClient *http.Client) bool {
+	isRetriable := func(err error) bool {
+		if net.IsConnectionReset(err) || net.IsConnectionRefused(err) {
+			return true
+		}
+
+		return false
+	}
+	err := retry.OnError(retry.DefaultBackoff, isRetriable, func() error {
+		resp, err := httpClient.Get(fmt.Sprintf("http://localhost:%d/healthz/ready", envoyHealthCheckPort))
+		if err != nil {
+			log.Log.Reason(err).V(4).Info("error when checking for istio-proxy presence")
+			return err
+		}
+
+		defer resp.Body.Close()
+		if resp.Header.Get("server") == "envoy" {
+			return nil
+		}
+		return fmt.Errorf("received response from non-istio health server: %s", resp.Header.Get("server"))
+	})
+	if err != nil {
+		return false
+	}
+
+	return true
+}
+
+func findPid(commandNamePrefix string) (int, error) {
+	entries, err := filepath.Glob("/proc/*/cmdline")
+	if err != nil {
+		return 0, err
+	}
+
+	for _, entry := range entries {
+		// #nosec No risk for path injection. Reading specific entries under /proc
+		content, err := ioutil.ReadFile(entry)
+		if err != nil {
+			return 0, err
+		}
+
+		if !strings.Contains(string(content), commandNamePrefix) {
+			continue
+		}
+
+		//   <empty> /    proc     /    $PID   /   cmdline
+		// items[0] sep items[1] sep items[2] sep  items[3]
+		items := strings.Split(entry, string(os.PathSeparator))
+		pid, err := strconv.Atoi(items[2])
+		if err != nil {
+			return 0, err
+		}
+
+		// everything matched, hooray!
+		return pid, nil
+	}
+
+	return 0, fmt.Errorf("Process %s not found in /proc", commandNamePrefix)
+}
+
+func removeArg(slice []string, arg string) []string {
+	indexArg := indexOf(slice, arg)
+	if indexArg != -1 {
+		slice[indexArg] = slice[len(slice)-1]
+		return slice[:len(slice)-1]
+	}
+
+	return slice
+}
+
+func indexOf(slice []string, value string) int {
+	for i, current := range slice {
+		if current == value {
+			return i
+		}
+	}
+	return -1
+}
```

Comment:
I think you could just do it dynamically, by replacing the arguments in place. 

func removeArg(args []string, arg string) []string {
	i := 0
	for _, elem := range args {
		if elem != arg {
			args[i] = elem
			i++
		}
	}
	args = args[:i]

	return args
}

What do you think?